src/main/java/com/microsoft/aad/msal4j/ClientApplicationBase.java

```diff
@@ -180,33 +154,91 @@ protected static void validateNotEmpty(String name, Set<String> set) {
      *         {@link AuthenticationResult} of the call. It contains Access
      *         Token, Refresh Token and the Access Token's expiration time.
      */
-    public CompletableFuture<AuthenticationResult> acquireTokenByRefreshToken(String refreshToken, Set<String> scopes) {
+
+    public CompletableFuture<AuthenticationResult> acquireTokenByRefreshToken(String refreshToken,
+                                                                              Set<String> scopes) {
         validateNotBlank("refreshToken", refreshToken);
+        validateNotEmpty("scopes", scopes);
 
-        final MsalOAuthAuthorizationGrant authGrant = new MsalOAuthAuthorizationGrant(
-                new RefreshTokenGrant(new RefreshToken(refreshToken)), scopes);
+        RefreshTokenRequest refreshTokenRequest = new RefreshTokenRequest(
+                refreshToken,
+                scopes,
+                clientAuthentication,
+                new RequestContext(clientId, correlationId));
 
-        return this.acquireToken(authGrant, clientAuthentication);
+        return this.InitializeRequest(refreshTokenRequest);
     }
 
-    AuthenticationResult acquireTokenCommon(AbstractMsalAuthorizationGrant authGrant, ClientAuthentication clientAuth,
-                                            ClientDataHttpHeaders headers) throws Exception {
+    CompletableFuture<AuthenticationResult> InitializeRequest(
```

Comment:
this looks more like execute request, initialize typically means prepare

Reply:
Good point, changed it.